src/extensions/surveys.tsx

```diff
@@ -36,10 +36,15 @@ const handleWidget = (posthog: PostHog, survey: Survey) => {
     Preact.render(<FeedbackWidget key={'feedback-survey'} posthog={posthog} survey={survey} />, shadow)
 }
 
+const visibleSurveys = new Set()
+
 export const callSurveys = (posthog: PostHog, forceReload: boolean = false) => {
     posthog?.getActiveMatchingSurveys((surveys) => {
         const nonAPISurveys = surveys.filter((survey) => survey.type !== 'api')
         nonAPISurveys.forEach((survey) => {
+            if (visibleSurveys.has(survey.id)) {
+                return // skip if survey is already visible
```

Comment:
FYI, I added this bit of code to prevent re-rendering the popup if this survey is already present on the page; I was noticing this weird behavior where we kept trying to render the survey afresh every few seconds.  This change mitigates it, although it might have consequences that I don't fully understand.  My UI tests seemed to work great, and it doesn't _look_ like I regressed anything, but I wanted to call this out in case it was dangerous.